lib/Auth/Tiqr.php

```diff
@@ -370,10 +370,13 @@ protected static function _processLogin($userId, $response, $sessionKey)
                         } else {
                             // Block user and destroy secret
                             $store->setBlocked($userId, true);
-                            $store->setSecret($userId, NULL);
                             $store->setLoginAttempts($userId, 0);
                             
-                            if ($tempBlockDuration > 0) {
+                            if (0 == $tempBlockDuration) {
+                                $store->setSecret($userId, NULL);
+                                $store->setBlocked($userId, true);
+                            }
```

Comment:
This last line is not needed here since it's done on line 372 as well. If you remove that, this PR can be merged